### PR TITLE
Prefer HTTPS if available for package sources

### PR DESCRIPTION
**Pull request by dcousens**
This pull request changes a number of `PKG_SOURCE_URL`s that were using `http:`, to use `https:` if they were available.

I understand that this shouldn't impact the integrity of the downloaded package, which has a `PKG_HASH` for that purpose.
This is none-the-less a beneficial change as HTTPS becomes a modern baseline.

### CODE REVIEW

File: package/utils/bsdiff/Makefile [hunk 1]
```diff
@@ -27,15 +27,15 @@ define Package/bsdiff
   CATEGORY:=Utilities
   DEPENDS:=+libbz2
   TITLE:=Binary diff tool
-  URL:=http://www.daemonology.net/bsdiff/
+  URL:=https://www.daemonology.net/bsdiff/
```
Comment (dcousens) [inline package/utils/bsdiff/Makefile:30]: This isn't a package source, but consistency

**Comment by dcousens** [comment]
Resolved commit formalities

**Comment by Ansuel** [comment]
@dcousens thanks for the pr. Commit description are mandatory so pls add it. Extra bonus point if you put in the description the update packages :D

**Comment by Ansuel** [comment]
Also hope it's not a problem but currently we don't accept github no-replay email :(

**Comment by robekl** [comment]
This will break transparent http caching

**Comment by Ansuel** [comment]
@robekl can you elaborate? majority of packet have https url also we are starting to check certificate.

**Comment by dcousens** [comment]
@Ansuel updated description and signed off with a different email (that poor inbox)

**Comment by dcousens** [comment]
Hopefully third time :pray:

Had to disable a few GitHub options to make this work

**Comment by Ansuel** [comment]
@dcousens what options you had to disable o.O

**Comment by dcousens** [comment]
Sorry, 1 option, the other option wasn't relevant in the end.

>  Block command line pushes that expose my email

Didn't have to use that when using `+openwrt`, but to fix the authorship error I needed to use an associated email to the account.
Ah well, that inbox is practically blocked nowadays anyway, non-stop phishing.

**Comment by diizzyy** [comment]
Unless sites actually redirects to (forces) https this is just for the sake of it and doesn't provide any benefit, it's been discussed many times before.

**Comment by Ansuel** [comment]
@diizzyy we now check the certificate as an additional security measure so it would make sense... even just for consistency. Are you against it or just neutral? I would honestly merge this and be done with it.

**Comment by dcousens** [comment]
> Unless sites actually redirects to (forces) https 

This changes the download to use `https`, there needn't be any redirection.
You might find some `http` routes are redirect to `https`, but why rely on that if we _know_ https is available?

**Comment by diizzyy** [comment]
@Ansuel 
By that you mean that it's validated and not ignored? That might bite you since OpenWrt tends to "support" very old hosts for a very long time.

**Comment by Ansuel** [comment]
@diizzyy with this https://github.com/openwrt/openwrt/pull/4948 we don't ignore https cert anymore. The option can be disabled from menuconfig but this is enabled by default.

**Comment by diizzyy** [comment]
Yeah, jow- also raised concerns about that but oh well...

**Comment by Ansuel** [comment]
@diizzyy true but if you think about it on a different view, host with expired cert should be replaced and if possible fixed. Also it will permit us to discover if our own cdn have problems with some certificate.
It can cause problems but ignoring the cert looks to be more a workaround than an intended thing.
(didn't want to start a discussion just giving you some reason why i merged that)

**Comment by diizzyy** [comment]
As much as we can argue about that OpenWrt have a history for keeping support for "legacy" platforms for a very long time so I'm just saying that it may clash with how releases are handled and supported.

**Comment by dcousens** [comment]
@diizzyy if we added fallbacks to `http:`, would that alleviate your concerns?
Without this change, I couldn't actually download these resources as the network environment I am in blocks HTTP outbound, only allowing HTTPS.

**Comment by hauke** [comment]
I do not understand the problem with using https by default?
If the original server is not available any more or the certificate is broken the build system will use one of our own mirrors automatically.
If you have an old build host with an outdated trust store you can run into problems, but distributions normally update it for systems still under support.

**Review by hauke — APPROVED** [review]

**Comment by diizzyy** [comment]
@dcousens 
That doesn't help anything at all just adding more workarounds.

@hauke
It would make more sense in that regard to add an option to opt-in regarding certificates.

That being said, I don't really care but I do see how this can cause breakage.

**Pull request merged**